course_info/static/course_info/js/controllers/PeopleController.js

```diff
@@ -14,110 +14,157 @@
+                    // the user already has an enrollment in the course
+                    $scope.messages.warnings.push({
+                        type: 'alreadyInCourse',
+                        // just pick the first enrollment to find the name
+                        name: $scope.getProfileFullName(filteredResults[0]),
+                        memberships: memberRecordsInCourse,
+                        searchTerm: searchTerm
+                    });
+                }
+            } else if (filteredResults.length == 0) {
+                // didn't find any people for the search term
+                $scope.messages.warnings.push({
+                    type: 'notFound',
+                    searchTerm: searchTerm
+                });
+            } else {  // if (filteredResults.length > 1)
+                // multiple profiles found for search term, do not add
+                $scope.messages.warnings.push({
+                    type: 'multipleProfiles',
+                    searchTerm: searchTerm,
+                    // just pick the first one to find the name
+                    name: $scope.getProfileFullName(filteredResults[0]),
+                    profiles: filteredResults
+                });
             }
-            else { // $scope.searchResults.length > 1
-                if ($scope.selectedResult.id) {
-                    $scope.addUserToCourse(searchTerm,
-                                           {user_id: $scope.selectedResult.id,
-                                            role_id: $scope.selectedRole.roleId});
+            $scope.tracking.failures++;
+        };
+        $scope.addNewMemberToCourse = function(userPostParams, userName,
+                                               searchTerm) {
+            /* Adds a single person to the course; handles the actual posting of
+             member info to the course via the API. Returns a promise for the
+             POST call response.
+             */
+            var handlePostSuccess = function(response) {
+                $scope.tracking.successes++;
+                return response;
+            };
+
+            var handlePostError = function(response) {
+                $scope.handleAjaxErrorResponse(response);
+                if ((((response||{}).data||{}).detail||'').indexOf(
+                            'Canvas API error details') != -1) {
+                    // There was a partial error (we caught
+                    // a Canvas API error). The user has been added to the
+                    // coursemanager db, but could not be added to Canvas.
+                    // In this case it's possible that the user will be added
+                    // during the next Canvas sync. We let the user know about
+                    // the partial failure and that it may correct itself.
+                    $scope.messages.warnings.push({
+                        type: 'partialFailure',
+                        failureDetail: response.data.detail,
```

Comment:
No, it doesn't, because we're only in here if `response.data.detail` exists and contains the string we're testing for.